src/main/java/org/folio/gobi/LookupService.java

```diff
@@ -127,6 +127,7 @@ public CompletableFuture<Organization> lookupOrganization(String vendorCode) {
     logger.debug("lookupOrganization:: Trying to look up organization by vendorCode: {}", vendorCode);
     String query = HelperUtils.encodeValue(String.format(CQL_CODE_STRING_FMT+CHECK_ORGANIZATION_ISVENDOR, vendorCode));
     String endpoint = String.format(GET_ORGANIZATION_ENDPOINT + QUERY, query);
+    ORGANIZATION_NAME = vendorCode;
```

Comment:
This approach will not work in case of multiple simultaneous threads because we have shared variable here.
So one thread can specify value of ORGANIZATION_NAME varaible and second parallel thread also invoked lookupOrganization and ovverrides value of this varaible, so the first initial value will be erased.

Reply:
fixed . Please review .